src/nodes/parameter.h

```diff
@@ -109,6 +112,11 @@ class ParameterBase : public Serialisable<>
     virtual bool assign(ParameterBase *other) = 0;
     // Return whether this parameter accepts the output type of the other
     virtual bool acceptsOutput(ParameterBase *other) const = 0;
+    // The type's representation as a raw int (only valid for int and enum)
+    virtual int asInt() const { return -1; }
+    // Set type's repreentation as a raw int (only valid for int and enum)
+    virtual void asInt(int value) { return; }
```

Comment:
I am slightly perturbed that the "rigidity" of the new `ParameterBase` implementation means that we have to do this, but I don't see a way around it at present (at least not without rewriting `EnumOptionsBase`). I would suggest the following name changes though for clarity.
```suggestion
    // The type's representation as a raw int (only valid for int and enum)
    virtual int getAsInt() const { return -1; }
    // Set type's representation as a raw int (only valid for int and enum)
    virtual void setFromInt(int value) { return; }
```